cms/lib/test/test_ai_aside_summary_config.py

```diff
@@ -0,0 +1,50 @@
+"""
+Tests for AiAsideSummaryConfig class.
+"""
+
+
+from unittest import TestCase
+from unittest.mock import patch
+
+from opaque_keys.edx.keys import CourseKey, UsageKey
+
+from cms.lib.ai_aside_summary_config import AiAsideSummaryConfig
+
+
+class AiAsideSummaryConfigTestCase(TestCase):
+    """ Tests for AiAsideSummaryConfig. """
+    AI_ASIDE_CONFIG_API_LOCATION = 'ai_aside.config_api.api'
+    COURSE_KEY = CourseKey.from_string("course-v1:test+Test+AiAsideSummaryConfigTestCase")
+    UNIT_KEY = UsageKey.from_string("block-v1:test+Test+AiAsideSummaryConfigTestCase+type@vertical+block@vertical_test")
+
+    def test_is_enabled(self):
+        """
+        Check if summary configuration is enabled using the ai_aside lib.
+        """
+        ai_aside_summary_config = AiAsideSummaryConfig(self.COURSE_KEY)
+        with patch(f'{self.AI_ASIDE_CONFIG_API_LOCATION}.is_summary_config_enabled', return_value=True):
+            self.assertTrue(ai_aside_summary_config.is_enabled())
+
+        with patch(f'{self.AI_ASIDE_CONFIG_API_LOCATION}.is_summary_config_enabled', return_value=False):
```

Comment:
we have data driven testing - ddt - available in edx-platform which is a better way to write these

you can run the same test twice for true and false with one decorator - here's an example:
https://github.com/openedx/edx-platform/blob/7e23feeb33861c0de3572364cd103c3929bb0588/lms/djangoapps/ora_staff_grader/tests/test_serializers.py#L633
this runs twice, using the boolean to set up test data and then to check it. DDT can get a lot fancier

note that as well as the ddt data decorator on that function  you also need a ddt.ddt decorator on the whole class

Reply:
I did not use ddt due the migrations is in a plugin, and the plugin itself you have the test ready. The `cms` should be trust that we are calling to the right methods itself